micro_sam/util.py

```diff
@@ -84,48 +49,82 @@
+            "vit_t": "6dbb90523a35330fedd7f1d3dfc66f995213d81b29a5ca8108dbcdd4e37d6c2f",
+            # first version of finetuned models on zenodo
+            "vit_h_lm": "9a65ee0cddc05a98d60469a12a058859c89dc3ea3ba39fed9b90d786253fbf26",
+            "vit_b_lm": "5a59cc4064092d54cd4d92cd967e39168f3760905431e868e474d60fe5464ecd",
+            "vit_h_em": "ae3798a0646c8df1d4db147998a2d37e402ff57d3aa4e571792fbb911d8a979c",
+            "vit_b_em": "c04a714a4e14a110f0eec055a65f7409d54e6bf733164d2933a0ce556f7d6f81",
+        },
+        # Now specify custom URLs for some of the files in the registry.
+        urls={
+            # the default segment anything models
+            "vit_h": "https://dl.fbaipublicfiles.com/segment_anything/sam_vit_h_4b8939.pth",
+            "vit_l": "https://dl.fbaipublicfiles.com/segment_anything/sam_vit_l_0b3195.pth",
+            "vit_b": "https://dl.fbaipublicfiles.com/segment_anything/sam_vit_b_01ec64.pth",
+            # the model with vit tiny backend fom https://github.com/ChaoningZhang/MobileSAM
+            "vit_t": "https://owncloud.gwdg.de/index.php/s/TuDzuwVDHd1ZDnQ/download",
+            # first version of finetuned models on zenodo
+            "vit_h_lm": "https://zenodo.org/record/8250299/files/vit_h_lm.pth?download=1",
+            "vit_b_lm": "https://zenodo.org/record/8250281/files/vit_b_lm.pth?download=1",
+            "vit_h_em": "https://zenodo.org/record/8250291/files/vit_h_em.pth?download=1",
+            "vit_b_em": "https://zenodo.org/record/8250260/files/vit_b_em.pth?download=1",
+        },
+    )
+    # This extra dictionary is needed for _get_checkpoint to download weights from a specific checkpoint
+    models.download_names={
+            "vit_h": "vit_h.pth",
+            "vit_l": "vit_l.pth",
+            "vit_b": "vit_b.pth",
+            "vit_t": "vit_t_mobile_sam.pth",
+            "vit_h_lm": "vit_h_lm.pth",
+            "vit_b_lm": "vit_b_lm.pth",
+            "vit_h_em": "vit_h_em.pth",
+            "vit_b_em": "vit_b_em.pth",
+        }
+    return models
+
+
+def _get_checkpoint(model_name, checkpoint_path):
     if checkpoint_path is None:
-        checkpoint_url = _MODEL_URLS[model_type]
-        checkpoint_name = _DOWNLOAD_NAMES.get(model_type, checkpoint_url.split("/")[-1])
-        checkpoint_path = os.path.join(_CHECKPOINT_FOLDER, checkpoint_name)
+        checkpoint_url = models().urls[model_name]
+        checkpoint_name = models().download_names.get(model_name, checkpoint_url.split("/")[-1])
+        checkpoint_path = os.path.join(microsam_cachedir()/"models", checkpoint_name)
 
         # download the checkpoint if necessary
         if not os.path.exists(checkpoint_path):
-            os.makedirs(_CHECKPOINT_FOLDER, exist_ok=True)
-            _download(checkpoint_url, checkpoint_path, model_type)
+            os.makedirs(microsam_cachedir()/"models", exist_ok=True)
+            pooch.retrieve(url=checkpoint_url, known_hash=models().registry.get(model_name))
```

Comment:
Does `known_hash` need to be None here (i.e. unknown)? I don't quite understand how someone can do fine tuning of model weights, but have the hash stay unchanged (which is I think what would have to happen for what's written here to make any sense).

Reply:
Ok, I think I don't fully understand what `pooch` is doing. I will check this out and follow up.